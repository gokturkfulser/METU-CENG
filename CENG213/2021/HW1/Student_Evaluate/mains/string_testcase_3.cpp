#include <iostream>

#include "String.h"
//#include "../String.h"

/*
 * Case 3 : Default constructor; various functions on empty string; print.
 */
int main() {
    std::cout << "-> Creating a string with default constructor." << std::endl;

    String str;

    std::cout << "-> Trying to erase first two characters from the string and then printing the string." << std::endl;

    str.eraseCharacters(0, 2);
    str.print(true);

    std::cout << "-> Trying to erase all characters from the string and then printing the string." << std::endl;

    str.eraseAllCharacters();
    str.print(true);

    std::cout << "-> Printing the string." << std::endl;

    str.print(true);

    return 0;
}
