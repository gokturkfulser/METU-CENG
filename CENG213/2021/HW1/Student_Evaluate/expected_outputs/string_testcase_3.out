-> Creating a string with default constructor.
-> Trying to erase first two characters from the string and then printing the string.
[]
-> Trying to erase all characters from the string and then printing the string.
[]
-> Printing the string.
[]
